pandas/core/arrays/period.py

```diff
@@ -185,6 +333,55 @@ def _generate_range(cls, start, end, periods, freq, fields):
 
         return subarr, freq
 
+    @classmethod
+    def _concat_same_type(cls, to_concat):
+        freq = {x.freq for x in to_concat}
+        assert len(freq) == 1
+        freq = list(freq)[0]
+        values = np.concatenate([x._data for x in to_concat])
+        return cls._from_ordinals(values, freq=freq)
+
+    @property
+    def asi8(self):
+        return self._ndarray_values.view('i8')
+
+    # --------------------------------------------------------------------
+    # Data / Attributes
+    @property
+    def nbytes(self):
+        # TODO(DatetimeArray): remove
+        return self._data.nbytes
+
+    @cache_readonly
+    def dtype(self):
+        return PeriodDtype.construct_from_string(self.freq)
+
+    @property
+    def _ndarray_values(self):
+        # Ordinals
+        return self._data
+
+    @property
+    def freq(self):
+        """Return the frequency object if it is set, otherwise None"""
+        return self._freq
+
+    @property
+    def flags(self):
+        """Deprecated"""
+        # Just here to support Index.flags deprecation.
+        # could also override PeriodIndex.flags if we don't want a
+        # version with PeriodArray.flags
+        return self.values.flags
+
+    @property
+    def base(self):
+        return self.values.base
+
+    @property
+    def data(self):
+        return self.astype(object).data
```

Comment:
> I would deal with this in Index/PeriodIndex

Agreed.  IIRC this is currently done in IndexOpsMixin.

Isn't the point of `.data` to point to the underlying data that this might be a view on?  Realizing an object array seems weird for that